### PR TITLE
[docs] Migrate content to the new location

### PR DESCRIPTION
**Pull request by siriwatknp**
https://github.com/mui-org/material-ui/pull/30757 has been merged.

- updated monorepo to latest
- `LIB_VERSION` is now using x repo package.json

**Current**: https://deploy-preview-3730--material-ui-x.netlify.app/components/data-grid/
**New**: https://deploy-preview-3730--material-ui-x.netlify.app/x/react-data-grid/

---

For more details about the migration, check out this PR > https://github.com/mui-org/material-ui/issues/30091

### CODE REVIEW

File: docs/scripts/api/buildApi.ts [hunk 1]
```diff
@@ -53,10 +53,14 @@ const createProject = (options: CreateProgramOptions): Project => {
...
-    : ['./docs/pages/api-docs/data-grid'];
-  await Promise.all(
+  let outputDirectories = ['./docs/pages/api-docs/data-grid'];
+  if (FEATURE_TOGGLE.enable_product_scope) {
+    outputDirectories = ['./docs/pages/api-docs/data-grid', './docs/pages/x/api/data-grid'];
+  }
+  if (FEATURE_TOGGLE.enable_redirects) {
+    outputDirectories = ['./docs/pages/x/api/data-grid'];
+  }
+  await Promise.allSettled(
```
Comment (cherniavskii) [inline docs/scripts/api/buildApi.ts]:
```suggestion
  await Promise.all(
```
It has to throw if there's an issue, see https://github.com/mui-org/material-ui-x/pull/3808

Comment (siriwatknp) [inline docs/scripts/api/buildApi.ts]:
```suggestion
  await Promise.all(
```

**Comment by cherniavskii** [comment]
Changing language leads to 404 on deploy preview
https://deploy-preview-3730--material-ui-x.netlify.app/zh/x/react-data-grid/
Screenshot 2022-02-04 at 10 59 50

File: docs/scripts/api/buildApi.ts (hunk 1 above)
Comment (cherniavskii) [reply docs/scripts/api/buildApi.ts]: I don't get why it says it's outdated :D
I can't see this change in the code

**Comment by siriwatknp** [comment]
> Changing language leads to 404 on deploy preview https://deploy-preview-3730--material-ui-x.netlify.app/zh/x/react-data-grid/ Screenshot 2022-02-04 at 10 59 50
> 
> EDIT: probably [f57cf8e](https://github.com/mui-org/material-ui-x/commit/f57cf8e0e9100dddf34404c143d8e93e2df27aa8) needs to be reapplied

This is the expected behavior for the deploy preview, i18n only works on production.

File: docs/scripts/api/buildApi.ts (hunk 1 above)
Comment (siriwatknp) [reply docs/scripts/api/buildApi.ts]: me too 🥲. will do it manually, thanks a lot!

**Comment by cherniavskii** [comment]
We'll need to pull https://github.com/mui-org/material-ui/pull/30923

**Review by flaviendelangle — APPROVED** [review]
For me it's a good first version of the split.
It will allow us to start reworking the longest pages :+1:

**Review by cherniavskii — APPROVED** [review]

**Comment by fahadashiq12** [comment]
I am worried about the DataGrid and DataGrid Generator, are they compatible in React 18 version?

**Comment by flaviendelangle** [comment]
Not related to the current PR
But yes it should be compatible, we will probably bump the peer deps in the next release

**Pull request merged**